query-engine/prisma-models/src/zipper.rs

```diff
@@ -0,0 +1,32 @@
+use crate::{
+    psl::{parser_database::walkers::Walker, schema_ast::ast},
+    InternalDataModelRef,
+};
+
+// Invariant: InternalDataModel must not contain any Zipper, this would be a reference counting
+// cycle (memory leak).
+#[derive(Debug, Clone)]
+pub struct Zipper<I> {
+    pub id: I,
+    pub(crate) dm: InternalDataModelRef,
+}
+
+impl<I: PartialEq> PartialEq for Zipper<I> {
+    fn eq(&self, other: &Self) -> bool {
+        self.id.eq(&other.id)
+    }
+}
+
+impl<I: Copy> Zipper<I> {
+    pub fn walker(&self) -> Walker<'_, I> {
+        self.dm.schema.db.walk(self.id)
+    }
+
+    pub fn zip<J>(self, id: J) -> Zipper<J> {
+        self.dm.zip(id)
+    }
```

Comment:
Am I right to say this isn't used? Is this intentional?

Reply:
I'll try deleting and see what happens. If it isn't used it shouldn't exist.

Reply:
Looks like it was indeed not used, good catch!